aws_topology/stackstate_checks/aws_topology/flowlogs.py

```diff
@@ -0,0 +1,168 @@
+
+    def __init__(self, bucket_name, account_id, session, location_info, agent, log):
+        self.bucket_name = bucket_name
+        self.session = session
+        self.location_info = location_info
+        self.agent = agent
+        self.account_id = account_id
+        self.log = log
+
+    def collect_networkinterfaces(self):
+        nwinterfaces = {}
+        client = self.session.client('ec2')
+        for itf in client_array_operation(
+            client,
+            "describe_network_interfaces",
+            "NetworkInterfaces"
+        ):
+            nwinterfaces[itf["NetworkInterfaceId"]] = itf
+        return nwinterfaces
+
+    def read_flowlog(self, not_before):
+        nwinterfaces = self.collect_networkinterfaces()
+        client = self.session.client('s3')
+        region = client.meta.region_name
+        bucket_name = self._get_bucket_name()
+        to_delete = []
+        files_to_handle = []
+        for pg in client.get_paginator('list_objects_v2').paginate(
+            Bucket=bucket_name,
+            Prefix='AWSLogs/{act}/vpcflowlogs/{rgn}/'.format(
+                act=self.account_id,
+                rgn=region
+            )
+        ):
+            for itm in pg.get('Contents') or []:
+                key_regex = r"(?P<y>\d{4})(?P<m>\d{2})(?P<d>\d{2})T(?P<h>\d{2})(?P<mm>\d{2})Z"
+                pts = re.search(key_regex, itm.get("Key", ""))
+                if pts:
+                    pts = pts.groupdict()
+                    dt = datetime(
+                        int(pts["y"]), int(pts["m"]), int(pts["d"]), int(pts["h"]), int(pts["mm"]), 0
+                    ).replace(tzinfo=pytz.utc)
+                    if dt < not_before:
+                        to_delete.append({
+                            'Key': itm['Key']
+                        })
+                        if len(to_delete) > self.MAX_S3_DELETES:
+                            self._delete_files(client, bucket_name, to_delete)
+                            to_delete = []
+                    else:
+                        files_to_handle.append(itm['Key'])
+
+        self._delete_files(client, bucket_name, to_delete)
+        return self.process_files(client, bucket_name, files_to_handle, nwinterfaces)
+
+    def _delete_files(self, client, bucket_name, files):
+        for i in range(0, len(files), self.MAX_S3_DELETES):
+            try:
+                self.log.info(
+                    "Deleting {} files from S3 bucket {}".format(len(files[i : i + self.MAX_S3_DELETES]), bucket_name)
+                )
+                client.delete_objects(
+                    Bucket=bucket_name,
+                    Delete={"Objects": files[i : i + self.MAX_S3_DELETES], "Quiet": True},
+                )
+            except Exception as e:
+                self.log.exception(e)
+                self.agent.warning("CloudtrailCollector: Deleting s3 files failed")
+
+    def process_files(self, client, bucket_name, files, nwinterfaces):
```

Comment:
Separate the processing of files from the creating of components and relations. So `process_files` should return a list of relations.